### PR TITLE
network: add SuppressPrefixLength option to RoutingPolicyRule

### LINKED ISSUE #14724: network: add support for suppress_prefixlength and suppress_ifgroup in routing policy rules
**Is your feature request related to a problem? Please describe.**
systemd-networkd doesn't support these options at the moment. I'll be fair: I'm only really interested in `suppress_prefixlength`, and more specifically in [the way `wg-quick(8)` uses it](https://www.wireguard.com/netns/#improved-rule-based-routing) to allow traffic to non-default local routes to be exempt from the tunnel.

**Describe the solution you'd like**
Add options `SuppressPrefixLength` and `SuppressInterfaceGroup` to the `RoutingPolicyRule` section in `.network` files.

### PR DESCRIPTION
**Pull request by ncfavier**
Closes #14724.

Well, half of it: I realised systemd didn't seem to support "interface groups" (correct me if I'm wrong), so I gave up on `suppress_ifgroup`. I can add it anyway if you'd like, the code is essentially the same.

### CODE REVIEW

File: test/fuzz/fuzz-unit-file/directives.service [hunk 1]
```diff
@@ -885,6 +885,7 @@ StandardOutput=
 StateDirectory=
 StateDirectoryMode=
 Storage=
+SuppressPrefixLength=
```
Comment (yuwata) [inline test/fuzz/fuzz-unit-file/directives.service]: Please drop this. This file is for unit files of pid1, e.g., .service.

File: src/network/networkd-routing-policy-rule.c [hunk 2]
```diff
@@ -1056,6 +1069,43 @@ int config_parse_routing_policy_rule_family(
...
+        assert(section);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = routing_policy_rule_new_static(network, filename, section_line, &n);
+        if (r < 0)
+                return r;
+
+        r = safe_atoi(rvalue, &n->suppress_prefixlen);
```
Comment (yuwata) [inline src/network/networkd-routing-policy-rule.c]: Maybe, `safe_atou()`? Also, please check the value (0 ... 128?).

File: test/fuzz/fuzz-unit-file/directives.service (hunk 1 above)
Comment (ncfavier) [reply test/fuzz/fuzz-unit-file/directives.service]: I got confused by some of the other .network directives appearing in that file, although they don't seem to belong in .service (e.g. InvertRule). Should those be removed as well?

Comment (yuwata) [reply test/fuzz/fuzz-unit-file/directives.service]: Ah... Please also drop them. Thank you.

Comment (ncfavier) [reply test/fuzz/fuzz-unit-file/directives.service]: Actually, it seems like this file contains every directive (see 170dd5f). I'll leave it as is, maybe the file should be renamed.

Comment (yuwata) [reply test/fuzz/fuzz-unit-file/directives.service]: Hmm. I did not know the commit. Thanks. LGTM.

**Comment by yuwata** [comment]
LGTM now. But @ncfavier can I squash the commits on merge?

**Comment by ncfavier** [comment]
Sure!

**Comment by yuwata** [comment]
Ahh,,, I merged my PR and that made conflict on this. Could you rebase your commits instead of  merging the master? Otherwise, our CIs cannot run correctly. Sorry.

**Comment by ncfavier** [comment]
No problem.

**Comment by yuwata** [comment]
Thanks.

**Pull request merged**